packages/cli-kit/src/public/node/node-package-manager.ts

```diff
@@ -248,21 +249,36 @@ export async function usesWorkspaces(appDirectory: string): Promise<boolean> {
  * Given an NPM dependency, it checks if there's a more recent version, and if there is, it returns its value.
  * @param dependency - The dependency name (e.g. react)
  * @param currentVersion - The current version.
+ * @param refreshIfOlderThanSeconds - If the last check was done more than this amount of seconds ago, it will
+ * refresh the cache. Defaults to always refreshing.
  * @returns A promise that resolves with a more recent version or undefined if there's no more recent version.
  */
-export async function checkForNewVersion(dependency: string, currentVersion: string): Promise<string | undefined> {
-  outputDebug(outputContent`Checking if there's a version of ${dependency} newer than ${currentVersion}`)
+export async function checkForNewVersion(
+  dependency: string,
+  currentVersion: string,
+  {
+    cacheExpiryInHours = 0,
+  } = {},
+): Promise<string | undefined> {
+  const getLatestVersion = async () => {
+    outputDebug(outputContent`Checking if there's a version of ${dependency} newer than ${currentVersion}`)
+    return getLatestNPMPackageVersion(dependency)
+  }
+
+  const cacheKey: PackageVersionKey = `npm-package-${dependency}`
+  let lastVersion
   try {
-    const lastVersion = await getLatestNPMPackageVersion(dependency)
-    if (lastVersion && new SemVer(currentVersion).compare(lastVersion) < 0) {
-      return lastVersion
-    } else {
-      return undefined
-    }
+    lastVersion = await cacheRetrieveOrRepopulate(cacheKey, getLatestVersion, cacheExpiryInHours * 24 * 1000)
```

Comment:
We need to convert from hours to miliseconds, no? 🤔 

```suggestion
    lastVersion = await cacheRetrieveOrRepopulate(cacheKey, getLatestVersion, cacheExpiryInHours * 3600 * 1000)
```